python/Ganga/GPIDev/Credentials2/VomsProxy.py

```diff
@@ -0,0 +1,148 @@
+                voms_command += ':'
+                if self.initial_requirements.group:
+                    voms_command += '/%s' % self.initial_requirements.group
+                if self.initial_requirements.role:
+                    voms_command += '/%s' % self.initial_requirements.role
+        logger.debug(voms_command)
+        command = 'voms-proxy-init -pwstdin -out %s %s' % (self.location, voms_command)
+        logger.debug(command)
+        return_code = self.shell.system(command)
+        if return_code == 0:
+            logger.debug('Grid proxy %s created. Valid for %s', self.location, self.time_left())
+        else:
+            raise CredentialRenewalError('Failed to create VOMS proxy')
+
+    @property
+    def shell(self):
+        if self._shell is None:
+            self._shell = Shell()
+        return self._shell
+
+    def destroy(self):
+        self.shell.cmd1('voms-proxy-destroy -file %s' % self.location, allowed_exit=[0, 1])
+
+        if os.path.isfile(self.location):
+            os.remove(self.location)
+
+    @cache
+    def info(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -all -file %s' % self.location)
+        return output
+
+    def field(self, label):
+        # type: (str) -> str
+        line = re.search(r'^{0}\s*: (.*)$'.format(label), self.info(), re.MULTILINE)
+        return line.group(1)
+
+    @property
+    @cache
+    def identity(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file %s -identity' % self.location)
+        return output.strip()
+
+    @property
+    @cache
+    def vo(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file %s -vo' % self.location)
+        if status != 0:
+            return None
+        return output.split(':')[0].strip()
+
+    @property
+    @cache
+    def role(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file %s -vo' % self.location)
+        if status != 0:
+            return None  # No VO
+        vo_list = output.split(':')
+        if len(vo_list) <= 1:
+            return None  # No command after VO
+        return vo_list[1].split('/')[-1].split('=')[-1].strip()
+
+    @property
+    @cache
+    def group(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file %s -vo' % self.location)
+        if status != 0:
+            return None  # No VO
+        vo_list = output.split(':')
+        if len(vo_list) <= 1:
+            return None  # No command after VO
+        # TODO Make this support multiple groups and subgroups
+        group_role_list = vo_list[1].split('/')
+        if len(group_role_list) <= 2:
+            return None  # No group specified in command
+        return group_role_list[-1].strip()
+
+    @cache
+    def expiry_time(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file %s -timeleft' % self.location)
+        if status != 0:
```

Comment:
In the case of dealing with Unix commands which have the usual `True==1`, `False==0` definition inverted it is confusing reading:
```python
status = <some unix command>
if status:
    raise RuntimeError('command failed')
```
since normally an `if something` suggests truth and therefore success. Explicitly comparing against the `0` literal shows that it really is that value we care about, not some other definition of truth.